kolibri/plugins/perseus_viewer/assets/src/views/PerseusRendererIndex.vue

```diff
@@ -537,8 +537,20 @@
           answerState = JSON.parse(
             replaceImageUrls(JSON.stringify(answerState), this.perseusFileUrl),
           );
+          const widgetIds = this.itemRenderer.getWidgetIds();
+          // Because of a switch between the input-number and numeric-input widgets
+          // it seems it is possible for us to have a serialized state with keys
+          // that do not correspond to any widgets. We need to sanitize the state
+          // before restoring it.
+          const sanitizedQuestions = {};
+          for (const key of widgetIds) {
+            if (answerState.question[key]) {
+              sanitizedQuestions[key] = answerState.question[key];
+            }
+          }
+          answerState.question = sanitizedQuestions;
```

Comment:
I'm not sure if this is a nit or confusion - why is `question` (singular) now set to `sanitizedQuestions` (object, and presumably plural). I get at a high level what's happening here - clear out anything that's not a match between the answerState questions and the widgetIds. But I don't understand really what this replacement object is. (Maybe that's a bit besides the point)

Reply:
I can update it to 'santizedQuestion' to avoid the slight confusion here, although I think the fault lies with Perseus for calling this 'question' when it can indeed be multiple prompts at once.